drivers/net/slip.c

```diff
@@ -397,7 +400,9 @@ static void slip_iface_init(struct net_if *iface)
 
 	ethernet_init(iface);
 
+#if defined(CONFIG_NET_LLDP)
```

Comment:
No need for this ifdef as `net_lldp_set_lldpdu(iface);` will resolve to empty function if `CONFIG_NET_LLDP` is not set.

Reply:
But `include/net/ethernet.h` doesn't include `net/lldp.h` if `CONFIG_NET_LLDP` is not defined and if `net/lldp.h` is not included there is no function prototype for `net_lldp_set_lldpdu(iface)` causing an implicit function declaration warning.

Reply:
Hmm, indeed. Well, in that case lets do it like you suggested.